manualdirectdebit.php

```diff
@@ -317,3 +312,16 @@ function manualdirectdebit_civicrm_links($op, $objectName, $objectId, &$links, &
     }
   }
 }
+
+/**
+ * Implements hook_civicrm_alterContent().
+ */
+function manualdirectdebit_civicrm_alterContent(&$content, $context, $tplName, &$object) {
+  if ($tplName == 'CRM/Contact/Page/View/CustomData.tpl' && $context == "page") {
+    $path = CRM_ManualDirectDebit_ExtensionUtil::path() . '/js/mandateEdit.js';
+    if (file_exists($path)) {
+      $js = file_get_contents($path);
+      $content .= "<script type='text/javascript'>" . $js . "</script>";
```

Comment:
I found this to be kind of a strange way to add js to the form. Definitely not kosher :) 

Do you know about this way to do it?

```php
CRM_Core_Resources::singleton()->addScriptFile('uk.co.compucorp.manualdirectdebit', 'mandateEdit.js');
```

Also, if you do it with `CRM_Core_Resources::singleton()->addScriptFile()`, you might try doing it on the buildForm hook instead.

https://docs.civicrm.org/dev/en/latest/standards/javascript/#adding-javascript-files

Reply:
I`ve tried to add script by 'CRM_Core_Resources::singleton()->addScriptFile()' , and it`s doesn`t work in hook 'buildForm' and even in 'alterContent'. But I`ve found out that it was possible in hook 'pageRun'. So, I`ve used it as more appropriate way.

Reply:
Cool